Task2/src/main/java/com/epam/mentoring/services/FilePersonDAOIterator.java

```
@@ -0,0 +1,162 @@
+package com.epam.mentoring.services;
+
+import com.epam.mentoring.exceptions.PersonDAOException;
+import com.epam.mentoring.models.Person;
+
+import java.io.*;
+import java.util.Iterator;
+import java.util.NoSuchElementException;
+
+/** Inner Iterator class*/
+class FilePersonDAOIterator implements Iterator<Person>, AutoCloseable
+{
+
+    private final FileInputStream fileInputStream;
+    private final InputStreamReader inputStreamReader;
+    private final BufferedReader reader;
+    private Person nextPerson;
+    /**Field to save any exception occurred, because overriding doesn't expect to add new exception types */
+    private Exception exception;
+
+    public FilePersonDAOIterator(String filePath) throws PersonDAOException
+    {
+        try
+        {
+            fileInputStream = new FileInputStream(filePath);
+            inputStreamReader = new InputStreamReader(fileInputStream);
+            reader = new BufferedReader(inputStreamReader);
+            nextPerson = getNextPerson();
+        }
+        catch (FileNotFoundException e)
+        {
+            throw new PersonDAOException("Cannot read a person. The data file '" + filePath + "' is not found.", e);
+        }
+
+    }
+
+    public Exception getException()
+    {
+        return exception;
+    }
+    /** Checks if the exception exsists */
+    public boolean exceptionOccurred()
```

Why not to throw the exception when it occurred?